### ISSUE TITLE
about page price doesn't update after flexible price is approved

### ISSUE THREAD

**Issue by pdpinch** [mitodl/mitxonline#1006]
### Steps to Reproduce

(tested in Chrome) 
1. Create a fresh account. 
2. Visit a non-program about page, like https://rc.mitxonline.mit.edu/courses/course-v1:MITxT+8.01.1x/
3. Click the price link on the page, currently "[$49 (or half off if you ask nicely)](https://rc.mitxonline.mit.edu/courses/course-v1:MITxT+8.01.1x/mitx-mooc-financial-assistance-request/)"
4. enter a low enough salary to qualify for financial assistance: for this test course, less than $50k
5. (you don't need to submit documents)
6. If necessary, using an admin account, approve the financial aid request at https://rc.mitxonline.mit.edu/staff-dashboard/
7. with your learner account, return to the tab with the about page
8. refresh the about page

### Expected Behavior

The price in the price box should update to half off, $24.50. 

### Actual Behavior

- The price stays $49. 
- Also if you click through to checkout, your cart is empty 

### screencast

https://mit.zoom.us/rec/share/PdLr2DY_k2hk4JnZRhau0NT17uDcbvZ6SlU2Jym8qi1cOvMoGJpzzXfoDLuEBwLi.OiDdffK12pnFw2eA?startTime=1663358993000

**Comment by jkachel** [comment]
I don't seem to have permission to view the screencast, but I was not able to reproduce this on RC using the given course. I created a new account (apptest+20220919@jkachel.com) and submitted a FinAid record, which I then approved. After refreshing, both the Price card and the Enroll Now upsell dialog have the proper 50% discount and direct me to the cart appropriately. Enrolling in the course for audit gets me the upsell card, which also has the right price in it. Can you retest? The code in question here is pretty new so wondering if the test was done before it was deployed to RC.

**Comment by asadali145** [comment]
Not sure how much this relates to the current issue but I faced a similar kind of issue with the Cart. I added a course to the cart and then tried to add another course but the cart displayed the old course in the cart. But in my case, it wasn't consistent as some of the times it worked fine. Had discussions with Arslan and he wasn't facing this issue.

**Comment by jkachel** [comment]
@pdpinch quick ping to remind you about the video and/or retesting (as RC includes code changes that involve this code)

**Comment by pdpinch** [comment]
@jkachel https://mit.zoom.us/rec/share/jkflmdEg8vcaUKpAd7imIvAdLtdZ4cghn1q3oJRz6F5brNbYF9QsVvyll12pz1ih.VBC-ALgPl5xu3DN3

**Comment by jkachel** [comment]
Thanks! I ran through this in step with the screencast and I can sort of reproduce it:

- I wasn't able to reproduce this with the course in the screencast (MITxT+8.01.1x). I followed along with the screencast using a new account and applied for financial assistance. Approving the request and then refreshing the course details page worked (the price updated successfully), even if I had the original tab open with the course details page. (Clicking the Financial Assistance link opens the form in a new tab.) In this case, the price link changed to the post-discount price and the link target changed to add the course to the cart; all of that worked OK.
- I _was_ able to get some similar weirdness with a DEDP course. 
   - With [Data Analysis for Social Scientists](https://rc.mitxonline.mit.edu/courses/course-v1:MITx+14.310x/) and a fresh user account, I clicked the financial assistance link (which opened in a new tab) and filled out the form. I then approved it in another window, and used the new tab to go back to the homepage and open the course page again. This time, the Price card did not change, in either the new tab or the old tab.
   - I then realized that 8.01.1x didn't have an active product. I updated the existing product in RC to be active and it worked as expected. 
- I repeated the Data Analysis 14.310x test using the Test SPOC course (which is in a program as well) and did not have any issues with that. The prices did change. (Though the Enroll upsell card needs help - the discount I got made the price < $0 and so my listed price was negative. In addition, the Get Certificate button did not work. These seem like separate issues.)  

I did check using the test account noted in the screencast and that one seemed to be OK at this point - loading the page for MITxT+8.01.1x showed the correct price in the upsell cards and in the Price card.

So, I think there's maybe some weirdness where there's no (active) product for the course, but I'm not sure what to do there. I feel like we want learners to be able to apply for financial assistance, but we can't update the Price card view if there's no active product to calculate the actual price with. I have entered a separate issue for the upsell card issue.
